src/statsmodels/formula.jl

```diff
@@ -283,6 +287,20 @@ end
 
 const DEFAULT_CONTRASTS = DummyCoding
 
+_levels(x::Union{CategoricalArray, NullableCategoricalArray}) = levels(x)
```

Comment:
Yes. One issue is that `unique(::NullableCategoricalArray)` has to return `Nullable` elements to be consistent with the generic `AbstractArray` method. But of course we can unwrap these in our internal method.